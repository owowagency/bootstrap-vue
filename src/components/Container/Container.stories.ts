import Col from '@/components/Col';
import Container from '.';
import Row from '@/components/Row';

export default {
    title: 'Layout/Container',
    component: Container,
};

const template = (args) => ({
    components: {Col, Container, Row},
    setup: () => ({args}),
    template: `
        <Container v-bind="args">
            <Row>
                <Col
                    v-for="i in 12"
                    class="border bg-light"
                >
                    {{ i }}
                </Col>
            </Row>
        </Container>
    `,
});

export const Default = template.bind({});

export const Fluid = template.bind({});

Fluid.args = {
    fluid: true,
};
