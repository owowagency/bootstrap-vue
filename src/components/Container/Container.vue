<template>
    <div :class="classes">
        <!-- @slot Displays content inside the container -->
        <slot />
    </div>
</template>

<script lang="ts" setup>
import {computed} from 'vue';
import useClasses from '@/composables/useClasses';

const props = defineProps({
    /**
     * Uses the `container-fluid` class
     */
    fluid: {
        type: Boolean,
        default: false,
    },
});

const {classes} = useClasses(computed(() => [
    `container${props.fluid ? '-fluid' : ''}`,
]));
</script>

<docs>
```vue
<Container>
    <Row>
        <Col>
            <h1>Hello World</h1>
        </Col>
    </Row>
</Container>
```
</docs>
