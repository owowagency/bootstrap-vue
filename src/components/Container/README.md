# Container

```vue
<Container>
    <Row>
        <Col>
            <h1>Hello World</h1>
        </Col>
    </Row>
</Container>
```

## Props

| Prop name | Description                      | Type    | Values | Default |
| --------- | -------------------------------- | ------- | ------ | ------- |
| fluid     | Uses the `container-fluid` class | boolean | -      | false   |

## Slots

| Name    | Description                           | Bindings |
| ------- | ------------------------------------- | -------- |
| default | Displays content inside the container |          |
