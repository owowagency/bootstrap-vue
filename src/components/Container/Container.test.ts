import Container from '.';
import {describe} from 'vitest';

describe('template', () => {
    componentRenderTest(Container);

    componentSlotRenderTest(Container);

    componentWrapperClassTest(Container, {fluid: true}, 'container-fluid');
});
