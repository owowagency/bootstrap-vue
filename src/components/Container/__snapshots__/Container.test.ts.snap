// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`template > renders default 1`] = `
"<div class=\\"container\\">
  <!-- @slot Displays content inside the container -->
</div>"
`;
